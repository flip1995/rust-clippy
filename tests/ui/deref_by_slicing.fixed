// run-rustfix

#![warn(clippy::deref_by_slicing)]

fn main() {
    let mut vec = vec![0];
    let _ = &*vec;
    let _ = &mut *vec;

    let ref_vec = &mut vec;
    let _ = &**ref_vec;
    let _ = &mut **ref_vec;

    let s = String::new();
    let _ = &*s;
}
