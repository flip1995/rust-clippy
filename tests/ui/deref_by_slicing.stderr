error: slicing when dereferencing would work
  --> $DIR/deref_by_slicing.rs:7:13
   |
LL |     let _ = &vec[..];
   |             ^^^^^^^^ help: dereference the original value instead: `&*vec`
   |
   = note: `-D clippy::deref-by-slicing` implied by `-D warnings`

error: slicing when dereferencing would work
  --> $DIR/deref_by_slicing.rs:8:13
   |
LL |     let _ = &mut vec[..];
   |             ^^^^^^^^^^^^ help: dereference the original value instead: `&mut *vec`

error: slicing when dereferencing would work
  --> $DIR/deref_by_slicing.rs:11:13
   |
LL |     let _ = &ref_vec[..];
   |             ^^^^^^^^^^^^ help: dereference the original value instead: `&**ref_vec`

error: slicing when dereferencing would work
  --> $DIR/deref_by_slicing.rs:12:13
   |
LL |     let _ = &mut ref_vec[..];
   |             ^^^^^^^^^^^^^^^^ help: dereference the original value instead: `&mut **ref_vec`

error: slicing when dereferencing would work
  --> $DIR/deref_by_slicing.rs:15:13
   |
LL |     let _ = &s[..];
   |             ^^^^^^ help: dereference the original value instead: `&*s`

error: aborting due to 5 previous errors

